examples/energy_mng/ui/widgets/pagination.slint

```diff
@@ -0,0 +1,43 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+import { Theme } from "../theme.slint";
+
+component Bubble inherits Rectangle {
+    in property <bool> selected;
+
+    width: 5px;
+    height: 5px;
```

Comment:
I don't think a components user should do that. It should be everything prepared as it's defined in the design. Otherwise you have a problem with the consistence in the implementation of the design. 

I used fixed sizes only for the bubbles and the menu button because that are special use cases with special behavior e.g. the menu button floats on the top of the menu and moves around. Not everything in an ui must be stretchable that the ui as whole is scalable. But most elements should be, yes.